include/vinecopulib/vinecop/implementation/tools_select.ipp

```diff
@@ -349,7 +335,7 @@ inline double
 VinecopSelector::get_next_threshold(std::vector<double>& thresholded_crits)
 {
   if (thresholded_crits.size() == 0) {
-    return 0.0;
```

Comment:
LOL, I really don't understand anything at this PR, sorry. Can you explain?

Reply:
- threshold = 1.0 for starting the search as explained above.
- `double mbicv_opt = std::numeric_limits<double>::max()` is because the independence model has mbicv > 0, so the check for `mbicv >= mbicv_opt` at the end of the iteration would also terminate the search after the first iteration. This fixes it.

Reply:
And yeah, this code isn't very pretty in general.